src/workspacer.Shared/Layout/GridLayoutEngine.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace workspacer
+{
+    public class GridLayoutEngine : ILayoutEngine
+    {
+        public GridLayoutEngine() { }
+
+        public string Name => "grid";
+
+        public IEnumerable<IWindowLocation> CalcLayout(IEnumerable<IWindow> windows, int spaceWidth, int spaceHeight)
+        {
+            var list = new List<IWindowLocation>();
+            var numWindows = windows.Count();
+
+            if (numWindows == 0)
+                return list;
+      
+
+                int rows = (int)(Math.Sqrt(numWindows) + 0.5);
+
+                while (numWindows % rows != 0)
+                {
+                    rows -= 1;
+                }
+
+                int cols = numWindows / rows;
```

Comment:
These lines seems to be indented 4 characters too much, making them appear as if they are part of the if-block.

Reply:
my bad, fixed in newest commit